docs/shared/checkpoint_learn_serialize.md

```diff
@@ -0,0 +1,118 @@
+\page ckpt_learn_serialize How to Serialize Data
+
+*Serialization* is the process of converting an object into a simple format
+that can be stored or transmitted and reconstructed later.
+checkpoint translates the object into a set of contiguous bits
+and provides the steps to reverse the process, i.e. to reconstitute
+the object from the set of bits.
+
+The extraction of the data from a series of bytes is called
+*deserialization*.
+
+\section serialize_builtin Serialization of built-in types
+
+This action is straightforward with the ```|``` operator.
+
+When the variable `a` has a built-in type and it needs to be serialized
+into the serializer object `s`, we simply write
+
+\code{.cpp}
+s | a;
+\endcode
+
+\section serialize_stl Serialization of STL containers
+
+The `|` operator has been overloaded for the STL containers:
+`std::array`, `std::deque`, `std::list`,
+`std::map`, `std::multimap`, `std::multiset`,
+`std::queue`,
+`std::set`, `std::string`,
+`std::unordered_map`, `std::unordered_multimap`,
+`std::unordered_multiset`, `std::unordered_set`,
+`std::vector`.
+
+The `|` operator is also overloaded for `std::pair`, `std::shared_ptr`, `std::unique_ptr`.
+
+When the variable `c` is such an STL object, whose template parameter(s) can
+be directly serialized into a serializer object `s`, we write
+
+\code{.cpp}
+s | c;
+\endcode
+
+\section serialize_class Serialization of custom class
+
+When a class (or structure) has to be serialized, the following steps are recommended:
+- define a default constructor;
+- define a templated function `template< class Serializer > void serialize(Serializer &s)` with
+  - a serialization step for all the member variables that need to be serialized.
+
+The example in \ref ckpt_learn_ex1 illustrates this situation.
+
+When the class (or structure) is not amenable to having a default constructor,
+the class must contains a static reconstructor method that serialization can use
+to construct the class.
+For a class to be serializable, it must have a valid reconstruction method
+(either a default constructor or a static reconstructor method).
+
+The example in \ref ckpt_learn_ex1 illustrates this situation.
+
+\section serialize_virtual Serialization of virtual class
+
+Consider the situation where the class `BaseVC` is abstract (i.e. it has a pure virtual method)
```

Comment:
There's nothing in the usage of these mechanisms that is specific to abstract classes and pure virtual methods. The root of the hierarchy can define every one of its methods, and this will all be as functional and desirable.

So, 

> Consider the situation where the class `BaseVC` is polymorphic (i.e. is has one or more virtual methods) and the class `CI` is derived from `BaseVC`

Reply:
Fixed